pkg/reconciler/common/prune.go

```diff
@@ -279,22 +283,54 @@ func (pruner *Pruner) checkAndCreate(ctx context.Context, uniquePruneNs, schedul
 				return err
 			}
 		}
-		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef)
+		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef, tC, computeTektonConfigSpecConfigHash)
 	}
 
 	// no change in config but the cronjob does not exist
 	if len(cronList.Items) == 0 && !configChanged {
-		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef)
+		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef, tC, computeTektonConfigSpecConfigHash)
 	}
 
 	// any case with config change
 	if configChanged {
-		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef)
+		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef, tC, computeTektonConfigSpecConfigHash)
 	}
+
+	if len(cronList.Items) > 0 {
+		for _, cronjob := range cronList.Items {
+			// get the cronjob to check previous annotations
+
+			cronjobName := cronjob.Name
+			annotations := cronjob.Annotations
+
+			nodeSelectorOrTolerationsChanged := checkTektonConfigSpecConfigHashChange(annotations, computeTektonConfigSpecConfigHash)
+			if nodeSelectorOrTolerationsChanged {
+				if err := pruner.deleteCronJob(ctx, cronjobName); err != nil {
+					return err
+				}
+				err := createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef, tC, computeTektonConfigSpecConfigHash)
+				if err != nil {
+					return err
+				}
+			}
+		}
+		return nil
+	}
+
 	return nil
 }
 
-func createCronJob(ctx context.Context, kc kubernetes.Interface, cronName, targetNs, pruneNs, schedule string, pruneContainers []corev1.Container, oR v1.OwnerReference) error {
+func checkTektonConfigSpecConfigHashChange(annotations map[string]string, currentComputedHash string) bool {
+	if len(annotations) != 0 {
```

Comment:
Here we are checking the annotation length, but this does not guarantee whether this key value will exist or not

Is not better to do like

```
lastHash, ok := annotations[v1alpha1.LastAppliedHashKey]
if ok {
      return lastHash != currentComputedHash
}
return true
```

Reply:
ok will change it.